src/NServiceBus.Core/DelayedDelivery/TimeoutManager/ProcessingFailureInfo.cs

```diff
@@ -4,15 +4,20 @@
 
     class ProcessingFailureInfo
     {
-        public ProcessingFailureInfo(int numberOfFailedAttempts, Exception exception)
+        public ProcessingFailureInfo(int numberOfFailedAttempts, Exception exception, bool shouldMoveToErrorQueue = false, bool shouldDeferForRetry = false)
```

Comment:
For now we are using separate `FailureStorage` instances so we could have only single flag. Something like `HandleOnNextProcessing'.